Documentation/writing-subplugin-tensor-filter.md

```diff
@@ -0,0 +1,79 @@
+# Writing a subplugin for NNStreamer's tensor\_filter
+
+You can support a new neural-network framework (e.g., Tensorflow and Caffe) or a new nerual-network hardware accelerator with its own software interface (e.g., openVINO for NCS and some python library for edge-TPU) by writing a tensor\_filter subplugin.
+
+It is called "**subplugin**" because it is a plugin for a GStreamer plugin, ```tensor_filter```.
+
+
+
+## Quick Guide on writing a tensor\_filter subplugin for a new framework/hardware.
+
+You can start writing a ```tensor_filter``` subplugin easily by using code-template/generator from nnstreamer-example.git. It is in ```/templates/tensor_filter_subplugin``` of ```nnstreamer-example.git```. The following is how to start writing a subplugin with the template for Tizen devices (5.5 M2 +). In this example, the target subplugin name is ```example```.
+
+```
+$ git clone https://github.com/nnsuite/nnstreamer-example.git
+...
+$ cd nnstreamer-example
+$ cd templates/tensor_filter_subplugin
+$ ./deploy example ~/example
+...
+...
+$ cd ~/example
+$ ls -l
+total 124
+-rw-rw-r--   1 babot babot    697 10-11 23:57 meson.build
+drwxrwxr-x   2 babot babot   4096 10-11 23:57 packaging
+drwxrwxr-x   2 babot babot   4096 10-11 23:57 src
+$ ls -l src/
+total 4
+-rw-rw-r-- 1 babot babot 3960 10-11 23:57 tensor_filter_subplugin.c
+$
+```
+
+Then, in ```src``` directory, you can fill in the callbacks.
+
+If you need to add dependencies for additional libraries of your own libraries, edit meson.build and packaging/*.spec accordingly as well.
```

Comment:
your own subplugin?

Reply:
Yes, it is about the reader's own subplugin to be written and the framework or hardware adaptation.